### PR TITLE
Add PERFORMANCE.md for examples and comparing to Oniguruma

### PR DESCRIPTION
**Pull request by robinst**
I learned all this while looking into https://github.com/google/fancy-regex/issues/14#issuecomment-360768175 and thought it might be good to have as documentation instead of a comment on an issue.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,6 +61,8 @@ guarantee. If an attacker can control the regular expressions that
 will be matched against, they will be able to successfully mount a
 denial-of-service attack. Be warned.
 
+See PERFORMANCE.md for some examples.
```
Comment (raphlinus) [inline README.md]: I don't think this will autolink, might be nice to add an explicit link.

Comment (robinst) [reply README.md]: 👍 done. Didn't know it was possible, but it's documented here: https://github.com/blog/1395-relative-links-in-markup-files

**Comment by raphlinus** [comment]
Thanks! I find it's always nice when you learn something in a review of a pull request :)

**Comment by robinst** [comment]
Thanks :). Have you seen #44? Would be good to get your thoughts on it!

**Pull request merged**